src/lib/es5.d.ts

```diff
@@ -1111,21 +1111,35 @@ interface JSON {
      * @param reviver A function that transforms the results. This function is called for each member of the object.
      * If a member contains nested objects, the nested objects are transformed before the parent object is.
      */
-    parse(text: string, reviver?: (this: any, key: string, value: any) => any): any;
+    parse(text: string, reviver?: (this: unknown, key: string, value: unknown) => unknown): any;
     /**
-     * Converts a JavaScript value to a JavaScript Object Notation (JSON) string.
-     * @param value A JavaScript value, usually an object or array, to be converted.
+     * Converts a JSON-representable value to a JavaScript Object Notation (JSON) string.
+     * @param value A JSON-representable value, usually an object or array, to be converted.
      * @param replacer A function that transforms the results.
      * @param space Adds indentation, white space, and line break characters to the return-value JSON text to make it easier to read.
      */
-    stringify(value: any, replacer?: (this: any, key: string, value: any) => any, space?: string | number): string;
+    stringify(value: boolean | number | string | object, replacer?: (this: unknown, key: string, value: unknown) => unknown, space?: string | number): string;
     /**
-     * Converts a JavaScript value to a JavaScript Object Notation (JSON) string.
-     * @param value A JavaScript value, usually an object or array, to be converted.
+     * Converts a JSON-representable value to a JavaScript Object Notation (JSON) string.
+     * @param value A JSON-representable value, usually an object or array, to be converted.
      * @param replacer An array of strings and numbers that acts as an approved list for selecting the object properties that will be stringified.
      * @param space Adds indentation, white space, and line break characters to the return-value JSON text to make it easier to read.
      */
-    stringify(value: any, replacer?: (number | string)[] | null, space?: string | number): string;
+    stringify(value: boolean | number | string | object, replacer?: (number | string)[] | null, space?: string | number): string;
+    /**
+     * Converts a JavaScript value to a JavaScript Object Notation (JSON) string or undefined.
+     * @param value A JavaScript value to be converted, which could potentially be a non-JSON-representable value.
+     * @param replacer A function that transforms the results.
+     * @param space Adds indentation, white space, and line break characters to the return-value JSON text to make it easier to read.
+     */
+    stringify(value: unknown, replacer?: (this: unknown, key: string, value: unknown) => unknown, space?: string | number): string | undefined;

```

Comment:
`this: unknown` requires `this` to be `unknown` or a supertype of unknown if it's specified. That doesn't sound right to me. What types are supposed to allowed and disallowed for `this` here?

Reply:
`this` is always set to the value being replaced, which doesn't have any distinct type. It could be a symbol or function or some other value added to the standard in the distant future, which is why I was conservative.

When you say "supertype of `unknown`"... I thought that `unknown` was the maximal type?